src/tiled/tilesetmodel.cpp

```diff
@@ -127,6 +136,50 @@ QMimeData *TilesetModel::mimeData(const QModelIndexList &indexes) const
     return mimeData;
 }
 
+bool TilesetModel::dropMimeData(const QMimeData *data, Qt::DropAction action,
+                  int row, int column,
+                  const QModelIndex &parent)
+{
+    Q_UNUSED(row);
+    Q_UNUSED(column);
+
+    if (!data || action != Qt::MoveAction)
+        return false;
+    if (!data->hasFormat(QLatin1String(TILES_MIMETYPE)))
+        return false;
+
+    QByteArray encodedData = data->data(QLatin1String(TILES_MIMETYPE));
+#if QT_VERSION < QT_VERSION_CHECK(6,0,0)
+    QDataStream stream(&encodedData, QIODevice::ReadOnly);
+#else
+    QDataStream stream(&encodedData, QDataStream::ReadOnly);
+#endif
+
+    int sourceId;
```

Comment:
Got a warning here that `sourceId` might be left uninitialized and still used. Would be better to make sure we only use it when it's set.

Reply:
Yeah, in the case there is no data embedded in the drop, it would be left uninitialized. The fix was really simple though, I've just didn't see it.